hack/pre-release.sh

```diff
@@ -0,0 +1,110 @@
+    stream_var=3
+  fi
+ 
+  # Locate and store the WMCO version from the makefile 
+  version=$(sed -n -e 's/^.*WMCO_VERSION ?= //p' Makefile)
+
+  # Increment the version 
+  updated_version=$(echo $version | 
+    awk -F. -v OFS=. -v stream_var="$stream_var" '{$stream_var += 1 ; print}')
+  
+  # Find the previous version from the olm.skipRange in cluster_service_version.yaml
+  last_version=$(grep "olm.skipRange" $cluster_service_version_yaml | 
+    grep -oP ">=\K([0-9]+\.[0-9]+\.[0-9]+)\s*<" | sed "s/\s*<//" | 
+    awk -F. -v OFS=. -v stream_var="$stream_var" '{$stream_var += 1 ; print}')
+
+  echo "WMCO version $version detected from the Makefile"
+
+  echo "Updating the Makefile to version $updated_version"
+  sed -i "s/$version/$updated_version/g" Makefile
+ 
+  echo "Updating the cluster service version yaml to $updated_version"
+  sed -i "s/$version/$updated_version/g" $cluster_service_version_yaml
+  
+  echo "Updating previous version to $version"
+  sed -i "s/$last_version/$version/g" $cluster_service_version_yaml
+
+  echo "Updating Dockerfile version to $updated_version"
+  sed -i "s/$version/$updated_version/g" bundle.Dockerfile 
+
+  echo "Updating annotation version to $updated_version"
+  sed -i "s/$version/$updated_version/g" bundle/metadata/annotations.yaml 
+
+  echo "Updating config-operator.package.yaml to $updated_version" 
+  sed -i "s/$version/$updated_version/g" bundle/windows-machine-config-operator.package.yaml
+}
+
+update_OCP_version() {
+  xy_stream=$1
+  cluster_service_version_yaml=config/manifests/bases/windows-machine-config-operator.clusterserviceversion.yaml
+  
+  if [[ $xy_stream = "x" ]]; then 
+    stream_var=2
+  else
+    exit 
+  fi 
+  
+  # Locate the OCP version from cluster_service_version.yaml
+  version=$(grep -o -E 'OCP \b([0-9]\.[0-9]{2})\b' $cluster_service_version_yaml | 
+    sed -E 's/OCP ([0-9]\.[0-9]{2})/\1/')
+
+  # Increment the version 
+  updated_version=$(echo $version | 
+    awk -F. -v OFS=. -v stream_var="$stream_var" '{$stream_var += 1 ; print}')
+
+  # Find the last version in the dockerfiles 
+  last_version=$(sed -n 's/.*\([0-9]\.[0-9][0-9]\).*/\1/p' build/Dockerfile)
+
+  echo "OCP version $version detected"
+
+  echo "Updating OCP version to $updated_version"
+  sed -i "s/$version/$updated_version/g" $cluster_service_version_yaml
+
+  echo "Updating Dockerfile version to $version"
+  sed -i "s/$last_version/$version/g" bundle.Dockerfile 
+
+  echo "Updating Dockerfile version to $version"
+  sed -i "s/$last_version/$version/g" build/Dockerfile
+
+  echo "Updating Dockerfile.base version to $version"
+  sed -i "s/$last_version/$version/g" build/Dockerfile.base
+
+  echo "Updating Dockerfile.ci version to $version"
+  sed -i "s/$last_version/$version/g" build/Dockerfile.ci
+
+  echo "Updating Dockerfile.wmco version to $version"
+  sed -i "s/$last_version/$version/g" build/Dockerfile.wmco
+}
+
+if [[ -z $1 ]]; then
+  while ! [[ $STREAM =~ ^(x|y|z)$ ]] 
```

Comment:
Are you providing an input if the user does not specify `x`, `y` or `z`? Do we really need this sort of nagging rather than just erroring out?

Reply:
```suggestion
  if [ -z $1 ]; then
  echo "Stream version not set"
  exit 1
  fi
```